network/dag/state.go

```diff
@@ -51,6 +50,7 @@ type state struct {
 	notifiers           map[string]Notifier
 	xorTree             *treeStore
 	ibltTree            *treeStore
+	highestLC           highestLC
```

Comment:
Mutex not ideal for atomic access. The classic solution was `atomic.Value`. Now with generics, I`d recommend `atomic.StorePointer` and  `atomic.LoadPointer` on this field—without the highestLC type— in combination with an explicit field name, such as atomicHighestLC.

Reply:
The current PR is a back port to the V5 branch. No point in changing it here, but sounds like a good change for the master branch

Reply:
Well there you go: #1565.